From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Subject: [PATCH] Fix build with external libexecinfo
Date: Mon,  5 Sep 2022 11:44:43 +0200

The backtracing functions may be implemented on a separate library instead
of in the libc, so try to find it.

Origin: Source Mage GNU/Linux
Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@iodev.co.uk>
---
 util/meson.build |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/util/meson.build b/util/meson.build
index 5cc209cc9288..54f0f8406f9f 100644
--- a/util/meson.build
+++ b/util/meson.build
@@ -60,5 +60,7 @@ foreach util : cairo_utils
 endforeach
 
 if conf.get('CAIRO_HAS_DLSYM', 0) == 1 and cc.has_header('execinfo.h')
-  libmallocstats = library('malloc-stats', 'malloc-stats.c', dependencies : dl_dep)
+  execinfo_dep = cc.find_library('execinfo', required: false)
+  libmallocstats = library('malloc-stats', 'malloc-stats.c',
+                           dependencies: [dl_dep, execinfo_dep])
 endif
